pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -201,6 +202,10 @@ type Image struct {
 
 	// Checksum is the checksum for the image.
 	Checksum string `json:"checksum"`
+
+	// ChecksumType is the checksum algorithm for the image.
+	// e.g md5, sha256, sha512
+	ChecksumType string `json:"checksumType,omitempty"`
```

Comment:
We should set up the kubebuilder enum validation rules for this field. The OperationalStatus and ErrorType fields show an example of how to do that. We should also declare the default for cases where the value is not provided.

Reply:
Fixed in the new updates.